osquery/tables/system/windows/processes.cpp

```diff
@@ -157,19 +161,39 @@ void genProcess(const WmiResultItem& result, QueryData& results_data) {
   s = result.GetLong("ProcessId", pid);
   r["pid"] = s.ok() ? BIGINT(pid) : BIGINT(-1);
 
+  if (context.isAnyColumnUsed(
+          {"elapsed_time", "handle_count", "percent_processor_time"})) {
+    std::map<std::string, std::int64_t> procPerfData;
+    procPerfData = perfData[pid];
+    r["elapsed_time"] = BIGINT(procPerfData["elapsed_time"]);
+    r["handle_count"] = BIGINT(procPerfData["handle_count"]);
+    r["percent_processor_time"] =
+        BIGINT(procPerfData["percent_processor_time"]);
+  }
+
   long uid = -1;
   long gid = -1;
   HANDLE hProcess = nullptr;
-  if (pid == currentPid) {
-    hProcess = GetCurrentProcess();
-  } else {
-    hProcess =
-        OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, false, pid);
-  }
 
-  if (GetLastError() == ERROR_ACCESS_DENIED) {
-    uid = 0;
-    gid = 0;
+  if (context.isAnyColumnUsed({"uid",
```

Comment:
My 2-cents on this -- I'd just leave this check out, and always at least make an attempt to get the `hProcess`. Considering the process handle is so fundamental to all of the subsequent calls we might make, and getting this handle takes a very small amount of time, I think it's fine to just leave it in regardless as there aren't many other columns one would want to query without getting a handle to the process.

Reply:
only 8 out of the 34 columns need a handle, i would take any optimization it can provide. what do you think

Reply:
Sure. I'm still of the opinion it's better to just have the process handle, as more often than not the API calls we'd be making will require it, but either way.